### PR TITLE
Updates breaking change documentation to use xrefs

### PR DESCRIPTION
**Pull request by conniey**
This PR:

* Changes all the usages I could from pointing to msdn.microsoft.com to the new docs.microsoft.com.
* Changes API references to use [cross-links](https://dotnet.github.io/docfx/tutorial/links_and_cross_references.html#define-uid)
* Edits the wording issues mentioned in PR 1923.
* Removes the [More Information] links that led to Application Compatibility page with the exact same info.

Required to generate correct data for https://github.com/dotnet/docs/pull/1923

/cc: @rpetrusha

### CODE REVIEW

**Comment by conniey** [comment]
@rpetrusha The [documentation for cross-referencing](https://dotnet.github.io/docfx/tutorial/links_and_cross_references.html#advanced-more-options-for-cross-reference) says that this is possible.  The `xrefmap.yml` map for possible cross references has an entry like:

```yml
- uid: System.Runtime.Serialization.NetDataContractSerializer.#ctor
  name: NetDataContractSerializer()
  href: api/System.Runtime.Serialization.NetDataContractSerializer.-ctor.html#System_Runtime_Serialization_NetDataContractSerializer__ctor
  commentId: M:System.Runtime.Serialization.NetDataContractSerializer.#ctor
  fullName: System.Runtime.Serialization.NetDataContractSerializer.NetDataContractSerializer()
  nameWithType: NetDataContractSerializer.NetDataContractSerializer()
```

So, if I wanted my text to display "System.Runtime.Serialization.NetDataContractSerializer.NetDataContractSerializer()", I'd use `displayProperty=fullName`, but if I just wanted to see "NetDataContractSerializer()", I'd use: `displayProperty=name`

File: Documentation/compatibility/a-concurrentdictionary-serialized-in-_net-4_5-with-netdatacontractserializer-cannot-be-deserialized-by-_net-4_5_1-or-4_5_2.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ Minor
 Available
 
 ### Change Description
-Due to internal changes to the type, `System.Collections.Concurrent.ConcurrentDictionary` objects that are serialized with the .NET Framework 4.5
-using the `NetDataContractSerializer` cannot be deserialized in the .NET Framework 4.5.1 or in the .NET Framework 4.5.2.
+Due to internal changes to the type, <xref:System.Collections.Concurrent.ConcurrentDictionary`2> objects that are serialized with the .NET Framework 4.5
+using the <xref:System.Runtime.Serialization.NetDataContractSerializer?displayProperty=name> cannot be deserialized in the .NET Framework 4.5.1 or in the .NET Framework 4.5.2.
 
 Note that moving the other direction (serializing with the .NET Framework 4.5.x and deserializing with the .NET Framework 4.5) works. Similarly,
```
Comment (conniey) [inline Documentation/compatibility/a-concurrentdictionary-serialized-in-_net-4_5-with-netdatacontractserializer-cannot-be-deserialized-by-_net-4_5_1-or-4_5_2.md]: Fixed

File: Documentation/compatibility/! Template.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 ### Scope
 [|Major|Minor|Edge|Transparent|]
-//A description of the Scopes can be found at https://msdn.microsoft.com/en-us/library/mt661864(v=vs.110).aspx
+//A description of the Scopes can be found at https://docs.microsoft.com/en-us/dotnet/articles/framework/migration-guide/net-compatibility-diagnostics
```
Comment (conniey) [inline Documentation/compatibility/! Template.md]: Fixed.

**Comment by rpetrusha** [comment]
@conniey, yes, that's how it should work. Is that how it's actually working? 
The displayProperty=fullName key/value should actually cause the text to be "NetDataContractSerializer.NetDataContractSerializer()" without the namespace name, although the current behavior is to include the fully qualified namespace as well.

**Comment by rpetrusha** [comment]
@conniey, actually, I apologize -- in your previous comment, I read what I thought was there, rather than what you actually wrote. The correct syntax is displayProperty=fullName. 
From

File: Documentation/compatibility/unexpected-invalidcastexception-from-invokemethod-activity-in-wf4.md [hunk 3]
```diff
@@ -13,16 +13,23 @@ Minor
...
 ### Recommended Action
-This behavior was reverted in a .NET Framework 4.5 servicing release. Please update the .NET Framework 4.5 (or upgrade to .NET Framework 4.5.1 or later) to fix the issue.
+
+This behavior was reverted in a .NET Framework 4.5 servicing release. Please
+update the .NET Framework 4.5 (or upgrade to .NET Framework 4.5.1 or later) to
+fix the issue.
 
 ### Affected APIs
-* `P:System.Activities.Statements.InvokeMethod.Parameters`
+* `P:<xref:System.Activities.Statements.InvokeMethod>.Parameters`
```
Comment (rpetrusha) [inline Documentation/compatibility/unexpected-invalidcastexception-from-invokemethod-activity-in-wf4.md]: This won't resolve to a valid URL target. It should be <xref:System.Activities.Statements.InvokeMethod.Parameters?displayProperty=fullName>
In general, affected APIs should all be fully qualified.

File: Documentation/compatibility/a-concurrentdictionary-serialized-in-_net-4_5-with-netdatacontractserializer-cannot-be-deserialized-by-_net-4_5_1-or-4_5_2.md [hunk 4]
```diff
@@ -13,20 +13,27 @@ Minor
...
 ### Change Description
-Due to internal changes to the type, `System.Collections.Concurrent.ConcurrentDictionary` objects that are serialized with the .NET Framework 4.5
-using the `NetDataContractSerializer` cannot be deserialized in the .NET Framework 4.5.1 or in the .NET Framework 4.5.2.
 
-Note that moving the other direction (serializing with the .NET Framework 4.5.x and deserializing with the .NET Framework 4.5) works. Similarly,
-all 4.x cross-version serialization works with the .NET Framework 4.6.
+Due to internal changes to the type,
+<xref:System.Collections.Concurrent.ConcurrentDictionary`2> objects that are
+serialized with the .NET Framework 4.5 using the
+<xref:System.Runtime.Serialization.NetDataContractSerializer?displayProperty=name>
```
Comment (rpetrusha) [inline Documentation/compatibility/a-concurrentdictionary-serialized-in-_net-4_5-with-netdatacontractserializer-cannot-be-deserialized-by-_net-4_5_1-or-4_5_2.md]: displayProperty=fullName

File: Documentation/compatibility/a-concurrentdictionary-serialized-in-_net-4_5-with-netdatacontractserializer-cannot-be-deserialized-by-_net-4_5_1-or-4_5_2.md [hunk 5]
```diff
@@ -13,20 +13,27 @@ Minor
...
+affected.
 
 - [ ] Quirked
 - [ ] Build-time break
 
 ### Recommended Action
-If it is necessary to serialize and deserialize a ConcurrentDictionary between the .NET Framework 4.5 and .NET Framework 4.5.1/4.5.2, an alternate serializer like the 
-DataContractSerializer or BinaryFormatter serializer should be used instead of the NetDataContractSerializer.
+If it is necessary to serialize and deserialize a <xref:System.Collections.Concurrent.ConcurrentDictionary`2?displayProperty=name> between the .NET Framework 4.5 and .NET Framework 4.5.1/4.5.2, an alternate serializer like the
+<xref:System.Runtime.Serialization.DataContractSerializer?displayProperty=name> or <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter?displayProperty=name> serializer should be used instead of the <xref:System.Runtime.Serialization.NetDataContractSerializer?displayProperty=name>.
```
Comment (rpetrusha) [inline Documentation/compatibility/a-concurrentdictionary-serialized-in-_net-4_5-with-netdatacontractserializer-cannot-be-deserialized-by-_net-4_5_1-or-4_5_2.md]: Each of these should be `displayProperty=fullName`

File: Documentation/compatibility/accessing-a-wpf-datagrid's-selected-items-from-a-handler-of-the-datagrid's-unloadingrow-event-can-cause-a-nullreferenceexception.md [hunk 6]
```diff
@@ -13,7 +13,15 @@ Minor
 Planned
 
 ### Change Description
-Due to a bug in the .NET Framework 4.5, event handlers for DataGrid events involving the removal of a row can cause a NullReferenceException to be thrown if they access the DataGrid's SelectedItem or SelectedItems properties.
+
+Due to a bug in the .NET Framework 4.5, event handlers for
+<xref:System.Windows.Controls.DataGrid> events involving the removal of a row
+can cause a <xref:System.NullReferenceException?displayProperty=name> to be
```
Comment (rpetrusha) [inline Documentation/compatibility/accessing-a-wpf-datagrid's-selected-items-from-a-handler-of-the-datagrid's-unloadingrow-event-can-cause-a-nullreferenceexception.md]: this line, and lines 22 and 24: ?displayProperty=fullName

File: Documentation/compatibility/ado_net-now-attempts-to-automatically-reconnect-broken-sql-connections.md [hunk 7]
```diff
@@ -10,13 +10,23 @@ Edge
...
 
 - [ ] Quirked
 - [ ] Build-time break
 
 ### Recommended Action
-If this feature is undesirable due to compatibility concerns, it can be disabled by setting the ConnectRetryCount property of a connection string (or SqlConnectionStringBuilder) to 0.
+
+If this feature is undesirable due to compatibility concerns, it can be disabled
+by setting the
+<xref:System.Data.SqlClient.SqlConnectionStringBuilder.ConnectRetryCount?displayProperty=name>
```
Comment (rpetrusha) [inline Documentation/compatibility/ado_net-now-attempts-to-automatically-reconnect-broken-sql-connections.md]: ?displayProperty=fullName

File: Documentation/compatibility/ado_net-now-attempts-to-automatically-reconnect-broken-sql-connections.md [hunk 8]
```diff
@@ -10,13 +10,23 @@ Edge
...
+property of a connection string (or
+<xref:System.Data.SqlClient.SqlConnectionStringBuilder?displayProperty=name>) to
```
Comment (rpetrusha) [inline Documentation/compatibility/ado_net-now-attempts-to-automatically-reconnect-broken-sql-connections.md]: ?displayProperty=fullName

File: Documentation/compatibility/ado_net-now-attempts-to-automatically-reconnect-broken-sql-connections.md [hunk 9]
```diff
@@ -10,13 +10,23 @@ Edge
...
+0.
 
 ### Affected APIs
 * `M:System.Data.IDbConnection.set_ConnectionString(System.String)`
```
Comment (rpetrusha) [inline Documentation/compatibility/ado_net-now-attempts-to-automatically-reconnect-broken-sql-connections.md]: It's best to do this as [IDbConnection.set_ConnectionString](<xref:System.Data.IDbConnection.ConnectionString>).

**Comment by conniey** [comment]
@rpetrusha

> "displayProperty: the **property** of display text when the cross reference is has resolved correctly.
e.g.: will be resolved as System.String."

The displayProperty=name maps correctly to just the name.  The **property** in your example is "fullname", the property I am trying to show is "name".

MD file: [eventlistener-truncates-strings-with-embedded-nulls.md](https://raw.githubusercontent.com/conniey/dotnet/c76aee9554bf809826fe8bbeb188009c9ce31d1c/Documentation/compatibility/eventlistener-truncates-strings-with-embedded-nulls.md) 
Preview: [46: EventListener truncates strings with embedded nulls](https://review.docs.microsoft.com/en-us/dotnet-new/articles/framework/migration-guide/runtime/4.5-4.5.1?branch=pr-en-us-1923#46-eventlistener-truncates-strings-with-embedded-nulls)

The ApiPort tool uses the Affected Apis.  We map those docIds to the correct cross reference when creating a PR for the docs site.

**Comment by conniey** [comment]
Another example is the event [System.Web.UI.WebControls.GridView.PageIndexChanging](https://review.docs.microsoft.com/en-us/dotnet-new/api/system.web.ui.webcontrols.gridview.pageindexchanging?branch=pr-en-us-1923&view=netframework-4.7), where I want to see just the name "PageIndexChanging" and use displayProperty=name:

[116: GridViews with AllowCustomPaging set to true may fire the PageIndexChanging event when leaving the final page of the view](https://review.docs.microsoft.com/en-us/dotnet-new/articles/framework/migration-guide/runtime/4.5-4.7?branch=pr-en-us-1923#116-gridviews-with-allowcustompaging-set-to-true-may-fire-the-pageindexchanging-event-when-leaving-the-final-page-of-the-view)

File: Documentation/compatibility/ado_net-now-attempts-to-automatically-reconnect-broken-sql-connections.md (hunk 9 above)
Comment (conniey) [inline Documentation/compatibility/ado_net-now-attempts-to-automatically-reconnect-broken-sql-connections.md]: We use documentIds. This is not a valid document ID.

**Comment by rpetrusha** [comment]
@conniey, I don't think that displayProperty=name is supported -- it happens to work since displaying the type or member name only is the default behavior. But since it's not breaking anything, it's harmless. I'll approve the changes, and leave it to you to merge.

**Pull request merged**